### ISSUE TITLE
Several Bugs Reporting

### ISSUE THREAD

**Issue by laserdark** [Madmanmayson/BattleInfoPlugin#4]
During the event, I found some bugs. Though I do wish to fix them alone, I'm scared whether I can clear all event maps for it lasts for only less than one week. So here comes the report.

- [ ] 1. Battle info only shows results up to the stage of pre-boss support expedition. It only occurs when the pre-boss expedition working, while the results of the boss node expedtion remain normal.

- [x] 2. Map route won't work correctly after the fight against abyssal combined fleets. Deleting the folder `..\BattleInfoPluginData` & restarting could temporarily ignore it once. Here's one exception that the surface task force(Suijou Dageki Butai) composition avoid the problem somehow.

- [x] 3. The damage of friend fleets do not be implemented.

PS: Is `SwfFormat.dll` no longer necessary as the cache transforms into HTML mode?

**Comment by Madmanmayson** [comment]
I'm just now working through E3 myself so I've only experienced the last 2 of the 3. The second one is usually caused by an unrecongnized part of the API result causing it to not realize combat has ended and therefor not proceed properly with resetting after battle (seems like its something new that loves to happen each event and I just now started looking into it for this one). Rebooting the program does force this to be fixed but as you said, only works for one run until it hangs again. 

I knew from last event friend fleets didn't work but I didn't get the data to implement them yet. Right now I'm doing a run so that I can get that data pulled and try to implement it. 

As for the first one, I've not used support expeds yet so I have no idea whats causing it or how to fix it yet but I'll try to investigate it this event as well though it might have to wait till the next one, assuming the problem persists.

Also SwfFormat was used to pull maps in the old version which of course doesn't work now that they are under a new api url and are pngs now but the program might still need it to avoid throwing an error. I've been avoiding messing with the menu that records the enemy formations and such but that's the only place I know its used.

**Comment by Madmanmayson** [comment]
As of right now Bug 1 hasn't been investigated but is likely being caused by the major changes in the api that occured a couple events ago which resulted in incorrect indices for enemy ships in fleets. If time and resources permit I'll try to get around to investigating runs with pre-boss support expeds. Progress has been made on the other 2 issues though.

Bug 2 has been fixed by temporarily disabling the checking of enemy fleet formations for the next until that system has been fixed (its had constant issues since major changes that occured to the combat API and I haven't looked into it at all since then). Fixed in commit d5539da.

Bug 3 is still being fully implemented but as of right now the window has been modified to show friend fleets when they appear in night battles in 50b9ce6. Basic damage support for them against enemy fleets has also been added in 8236d60 however damage to the friendly fleet is currently being reported on the user fleet instead. I'll try to have that part finished tomorrow.

**Comment by Madmanmayson** [comment]
As of now, friend support fleets should be fully implemented but final testing for those and checking the first bug reported will have to wait until next main event

**Comment by laserdark** [comment]
I seldom use support expedition either. However, as far as I know, map 5 is one area applied with support expedition, which makes it possible to check some details of the first bug.

**Comment by laserdark** [comment]
Here are some details for BUG 1. Bug 1 could be observed in regular maps as well. 
And the representation between two kinds of pre-boss expedition - normal pre-boss expedition & night battle pre-boss expedition is different.
For the normal battles, the problem has been described above.
With night battle pre-boss expedition(e.g. map 5-3), the damge of expedition would not be calculated into results, which looks like being caused by the api changes.

**Comment by laserdark** [comment]
It seems that something goes wrong in the function `Support`(line 46) in the file `..\Models\BattleData.Damage.cs`. When I comment out this function, the plugin would show data for a new battle, while it shows nothing at the release version.
I guess the key may in the function `GetDamages`, but there's no hints in debug mode.

PS: the damage from the friend fleets is still calucated to the self fleet, plz have a look

**Comment by Madmanmayson** [comment]
Yeah finally had a chance to test support damage and found that too, going to work on fixing that now and add support for the new ambush nodes. Hopefully both of those will be done by the end of the day

As for friend fleets I thought I got the damage flipped around after the last event ended however I never had any opportunities to test is as they aren't available in the main game afaik. It also looks like they are not available in this event either so I still can't work on seeing if they are still reversed.
